### PR TITLE
Migrate to Python3 and PyQt5

### PR DESCRIPTION
**Pull request by ptersilie**
Needs to be rebased once #265 is merged first.

With the imminent retirement of Python2 it is about time we move Eco over to Python3, which meant we also had to upgrade to PyQt5.

The first two commits are mostly mechanic. 7ec63da moves things over to PyQt5, which mostly fixes imports but also updates signals which have changed in PyQt5. b03b557 uses `2to3` to automatically migrate everything over to Python3 with a6eb240 adding some manual fixes necessary to work with Python3.

c0d3359 updates Travis to use Python3 and PyQt5. 
86aff81 fixes escaping in grammars (see example diff below).
b4a2408 fixes some deprecation warnings.
08dcad8 Fixes a variety of tests which stopped working due to either Python3 or some of the changes above.

Example diff for grammar fixes:
```diff
@@ -13,7 +13,7 @@ P ::= "INT"
 INT:"[0-9]+"
 plus:"\+"
 mul:"\*"
-<ws>:"[ \\t]+"
+<ws>:"[ \t]+"
 cmt_start:"/\*"
 cmt_end:"\*/"
-<return>:"[\\n\\r]"
\ No newline at end of file
+<return>:"[\n\r]"
\ No newline at end of file
```

### CODE REVIEW

**Comment by ltratt** [comment]
I'm not sure GitHub likes this PR -- it says you've added 0.6KLoC but deleted over 10KLoC!

**Comment by ptersilie** [comment]
I think it's because I based this branch off of #265 so this probably will go away once we rebase.

**Comment by ltratt** [comment]
OK, probably best to rebase now then.

**Comment by ptersilie** [comment]
Rebased.

**Comment by ltratt** [comment]
I must admit to mild nerves about this PR: I think we could easily mess many minor things up and not notice. For example, I loaded Eco up, pressed Ctrl+N, selected PHP, then PHP & Python, pressed OK and then got:

```
Traceback (most recent call last):
  File "eco.py", line 1148, in newfile
    etab.set_language(lang, lview.getWhitespace())
  File "/tmp/eco/lib/eco/editortab.py", line 121, in set_language
    incparser, inclexer = lang.load()
  File "/tmp/eco/lib/eco/grammars/grammars.py", line 94, in load
    pickle_id = self.pickleid(whitespace)
  File "/tmp/eco/lib/eco/grammars/grammars.py", line 174, in pickleid
    with open(self.filename, "r", encoding="latin-1") as f:
TypeError: 'encoding' is an invalid keyword argument for this function
```

File: lib/eco/dotviewer/graphdisplay.py [hunk 1]
```diff
@@ -47,14 +47,16 @@ def permute_mods(base, args):
...
+    def __init__(self, xxx_todo_changeme=(800,680)):
         # initialize the modules by hand, to avoid initializing too much
         # (e.g. the sound system)
+        (w,h) = xxx_todo_changeme
         pygame.display.init()
         pygame.font.init()
         self.resize((w,h))
 
-    def resize(self, (w,h)):
+    def resize(self, xxx_todo_changeme1):
```
Comment (ltratt) [inline lib/eco/dotviewer/graphdisplay.py]: Variable names like this scare me!

Comment (ptersilie) [reply lib/eco/dotviewer/graphdisplay.py]: This is from the `2to3` conversion. I forgot to rename those variables again.

**Review by ptersilie — COMMENTED** [review]
> I must admit to mild nerves about this PR: I think we could easily mess many minor things up and not notice. For example, I loaded Eco up, pressed Ctrl+N, selected PHP, then PHP & Python, pressed OK and then got:
> 
> ```
> Traceback (most recent call last):
>   File "eco.py", line 1148, in newfile
>     etab.set_language(lang, lview.getWhitespace())
>   File "/tmp/eco/lib/eco/editortab.py", line 121, in set_language
>     incparser, inclexer = lang.load()
>   File "/tmp/eco/lib/eco/grammars/grammars.py", line 94, in load
>     pickle_id = self.pickleid(whitespace)
>   File "/tmp/eco/lib/eco/grammars/grammars.py", line 174, in pickleid
>     with open(self.filename, "r", encoding="latin-1") as f:
> TypeError: 'encoding' is an invalid keyword argument for this function
> ```

Huh, are you running Python3?

**Comment by ltratt** [comment]
It turns out that `bin/eco` still uses `python2.7`! We should fix that ;)

**Comment by ptersilie** [comment]
I agree with you that we could easily mess minor things up, though I have tested most of the the major Eco features I could think off. I just don't have a better idea since testing editors is hard.

**Comment by ptersilie** [comment]
> It turns out that `bin/eco` still uses `python2.7`! We should fix that ;)

I'm surprised you even managed to get into a menu with Python2.

**Comment by ltratt** [comment]
OK, so I think we should fix those `xxx` variable names, but otherwise things seem to work OK under light testing.

**Comment by ptersilie** [comment]
See if those work for you.

**Comment by ltratt** [comment]
Please squash.

**Comment by ptersilie** [comment]
I also added another fix for the parse tree viewer. They both will need to be squashed before merging.

**Comment by ptersilie** [comment]
Squashed!

**Review by ltratt — APPROVED** [review]

**Pull request merged**